zebra-chain/src/primitives/address.rs

```diff
@@ -47,17 +50,34 @@ impl TryFrom<zcash_address::Network> for Network {
     fn try_from(network: zcash_address::Network) -> Result<Self, Self::Error> {
         match network {
             zcash_address::Network::Main => Ok(Network::Mainnet),
-            zcash_address::Network::Test => Ok(Network::Testnet),
+            zcash_address::Network::Test => Ok(Network::new_default_testnet()),
             zcash_address::Network::Regtest => Err("unsupported Zcash network parameters".into()),
         }
     }
 }
 
-impl From<&Network> for zcash_address::Network {
-    fn from(network: &Network) -> Self {
+impl TryFrom<&Network> for zcash_address::Network {
+    type Error = UnsupportedNetwork;
+
+    fn try_from(network: &Network) -> Result<Self, Self::Error> {
```

Comment:
This is inconsistent with the `From<NetworkKind>` impl below. Check if this is still needed with the next version of librustzcash at all. If it _is_ still needed, consider adding an `UnknownTestnet` variant of NetworkKind.